### PR TITLE
Cleanup rustdoc warnings

### PR DESCRIPTION
**Pull request by jyn514**
## Clean up error reporting for deprecated passes

Using `error!` here goes all the way back to the original commit, https://github.com/rust-lang/rust/pull/8540. I don't see any reason to use logging; rustdoc should use diagnostics wherever possible. See https://github.com/rust-lang/rust/pull/81932#issuecomment-785291244 for further context.

- Use spans for deprecated attributes
- Use a proper diagnostic for unknown passes, instead of error logging
- Add tests for unknown passes
- Improve some wording in diagnostics

##  Report that `doc(plugins)` doesn't work using diagnostics instead of `eprintln!`

This also adds a test for the output.

This was added in https://github.com/rust-lang/rust/pull/52194. I don't see any particular reason not to use diagnostics here, I think it was just missed in https://github.com/rust-lang/rust/pull/50541.

### CODE REVIEW

File: src/librustdoc/config.rs [hunk 1]
```diff
@@ -670,9 +670,8 @@ fn check_deprecated_options(matches: &getopts::Matches, diag: &rustc_errors::Han
             {
                 continue;
             }
-            let mut err =
-                diag.struct_warn(&format!("the '{}' flag is considered deprecated", flag));
-            err.warn(
+            let mut err = diag.struct_warn(&format!("the '{}' flag is deprecated", flag));
```
Comment (GuillaumeGomez) [inline src/librustdoc/config.rs:673]:
```suggestion
            let mut err = diag.struct_warn(&format!("the `{}` flag is deprecated", flag));
```

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks! r=me with green CI

**Comment by jyn514** [comment]
Note that this now adds `sym::plugins` as a known doc attribute; it was missing before.

**Comment by jyn514** [comment]
ping @GuillaumeGomez - I changed this a little since your last review, you may want to take another look. In particular this adds `sym::plugins` to the list of rustdoc attributes (it was missed in #82662).

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+

**Pull request merged**